lib/github.ml

```diff
@@ -235,6 +250,79 @@ let assert_tag_exists ~dry_run tag =
   if Vcs.tag_exists ~dry_run repo tag then Ok ()
   else R.error_msgf "%s is not a valid tag" tag
 
+(* Ask the user then push the tag. Guess the ssh URI from the dev-repo.
+   This function can abort:
+   - The user answered "N" to pushing the tag
+   - The push failed
+
+   This function does nothing if the tag is already present on the remote and
+   point to the same ref. *)
+let push_tag ~dry_run ~yes ~dev_repo vcs tag =
+  let remote_has_tag_uptodate () =
+    Vcs.commit_id ~dirty:false ~commit_ish:tag vcs >>= fun local_rev ->
+    Vcs.ls_remote ~dry_run vcs ~kind:`Tag ~filter:tag dev_repo >>= function
+    | [] -> Ok false
+    | (remote_rev_unpeeled, _) :: _ -> (
+        (* Resolve again in case of annotated commits (most common case).
+           This is a no-op for non-annotated commits. In case of error, we
+           can assume that the remote is different because we checked that we
+           have the tag locally. *)
+        match Vcs.commit_id ~commit_ish:remote_rev_unpeeled vcs with
+        | Ok remote_rev when remote_rev = local_rev ->
+            if remote_rev_unpeeled = remote_rev then
+              App_log.unhappy (fun l ->
+                  l
+                    "The tag present on the remote is not annotated (it was \
+                     not created by dune-release tag.)");
+            Ok true
+        | r ->
+            let pp_r fmt = function
+              | Ok remote_rev -> Text.Pp.commit fmt remote_rev
+              | Error _ -> Format.fprintf fmt "that we don't have locally"
+            in
+            App_log.unhappy (fun l ->
+                l
+                  "The tag %a is present on the remote but points to a \
+                   different commit (%a)."
+                  Text.Pp.version tag pp_r r);
+            Ok false )
+  in
+  remote_has_tag_uptodate () >>= function
+  | true ->
+      App_log.status (fun l ->
+          l "The tag %a is present and uptodate on the remote." Text.Pp.version
```

Comment:
Sorry to be so picky but could you change the message to:
```suggestion
          l "The tag %a is present and uptodate on the remote: skipping the tag push" Text.Pp.version
```
To clearly state what is going on here, otherwise it's unclear that anything is skipped from the regular workflow and new user might get confused. They might think it's some kind of check and that it's their responsibility to push the tag for instance.